src/components/date/date.js

```diff
@@ -180,6 +180,21 @@ class Date extends React.Component {
   }
 
   /**
+   * A lifecycle method for when the component is removed from the page.
+   *
+   * @method componentWillUnmount
+   * @return {void}
+   */
+  componentWillUnmount() {
+    /* istanbul ignore if */
+    if (super.componentWillUnmount) { super.componentWillUnmount(); }
```

Comment:
Sorry if this has been discussed before but when wouldn't a parent have a componentWillUnmount?

Is this an issue with tests

Reply:
Fair point @SeanArmstrong. TBH I was just copying the idiom from the other lifecycle functions as I'd not seen it before. Could the same be said of the other conditional calls in the file to the `super` lifecycle functions?

Reply:
@SeanArmstrong this is my mistake. I think I saw this conditional in the input validation Decorator which probably needs it as it doesn't inherit from `React.Component`. I had to add these two lifecycle functions to date, and have not removed the calls to `super`. I'll push an update.